packages/babel-helper-transform-fixture-test-runner/src/index.js

```diff
@@ -183,7 +183,13 @@ export default function (
           } else {
             if (task.exec.code) {
               const result = run(task);
-              if (result && typeof result.then === "function") {
+              if (
+                result &&
+                (
+                  typeof result.then === "function" ||
+                  typeof result.catch === "function"
```

Comment:
@existentialism I added separate test (with rejected promise), but new test works without this line (`typeof result.catch === "function"`). I think this is because the promises always have `.then`.